icon_validator/rules/plugin_validators/version_bump_validator.py

```diff
@@ -0,0 +1,345 @@
+        if SpecConstants.CONNECTIONS in remote:
+            if SpecConstants.CONNECTIONS not in local:
+                raise ValidationException("Connection removed without a major version increment."
+                                          f"{self.MAJOR_INSTRUCTIONS_STRING}")
+            for key, value in remote[SpecConstants.CONNECTIONS].items():
+                if key not in local[SpecConstants.CONNECTIONS]:
+                    raise ValidationException(f"{key} removed from connection without a major version increment."
+                                              f"{self.MAJOR_INSTRUCTIONS_STRING}")
+            for key, value in local[SpecConstants.CONNECTIONS].items():
+                if key not in remote[SpecConstants.CONNECTIONS] and value[SpecConstants.REQUIRED]:
+                    raise ValidationException(f"{key} added to connection as required input without major version"
+                                              f"increment.{self.MAJOR_INSTRUCTIONS_STRING}")
+            # established the same keys/vals at this point. Check titles / types next
+            for key, value in local[SpecConstants.CONNECTIONS].items():
+                if key in remote[SpecConstants.CONNECTIONS]:
+                    curr_compare = remote[SpecConstants.CONNECTIONS][key]
+                    if value[SpecConstants.TITLE] != curr_compare[SpecConstants.TITLE]:
+                        raise ValidationException(f"Title changed in connection field {key}, requiring a major version"
+                                                  f" increment.{self.MAJOR_INSTRUCTIONS_STRING}")
+                    if value[SpecConstants.TYPE] != curr_compare[SpecConstants.TYPE]:
+                        raise ValidationException(f"Type changed in connection field {key}, requiring a major version"
+                                                  f" increment.{self.MAJOR_INSTRUCTIONS_STRING}")
+        else:
+            if SpecConstants.CONNECTIONS in local:
+                raise ValidationException(f"Connection newly added to {RepoConstants.PLUGIN_SPEC}. This requires a"
+                                          f" major version increment.{self.MAJOR_INSTRUCTIONS_STRING}")
+
+    def validate_inner_fields(self, remote, local):
+        self.validate_no_sections_removed(remote, local)
+        if SpecConstants.INPUT in remote and SpecConstants.INPUT in local:
+            self.validate_no_input_removed(remote, local)
+            self.validate_no_input_new_or_required(remote, local)
+        if SpecConstants.OUTPUT in remote and SpecConstants.OUTPUT in local:
+            self.validate_no_output_removed(remote, local)
+            self.validate_no_output_no_longer_required(remote, local)
+
+        self.validate_no_titles_changed(remote, local)
+        self.validate_no_inner_type_changes(remote, local)
+
+    def check_major_version_increment_needed(self, remote, local):
+        local_version = local["version"].split('.')
```

Comment:
Check for hyphen semver strings with patch suffixes.

Reply:
Made a function that takes everything before a hyphen (if a hyphen is in the last slot)

Reply:
Added